testsuite/single_node_performance.py

```diff
@@ -74,16 +76,132 @@
 
 
 def execute_command(command):
-    try:
-        output = subprocess.check_output(
-            command, shell=True, text=True, cwd=target_directory
+    result = []
+    with Popen(
+        command,
+        shell=True,
+        text=True,
+        cwd=target_directory,
+        stdout=PIPE,
+        bufsize=1,
+        universal_newlines=True,
+    ) as p:
+        # stream to output while command is executing
+        for line in p.stdout:
+            print(line, end="")
+            result.append(line)
+
+    if p.returncode != 0:
+        raise CalledProcessError(p.returncode, p.args)
+
+    # return the full output in the end for postprocessing
+    return "\n".join(result)
+
+
+@dataclass
+class RunGroupKey:
+    transaction_type: str
+    module_working_set_size: int
+    executor_type: str
+
+
+@dataclass
+class RunResults:
+    tps: float
+    gps: float
+    fraction_in_execution: float
+    fraction_of_execution_in_vm: float
+    fraction_in_commit: float
+
+
+@dataclass
+class RunGroupInstance:
+    key: RunGroupKey
+    single_node_result: RunResults
+    concurrency_level_results: dict[int, RunResults]
+    block_size: int
+    expected_tps: float
+
+
+def extract_run_results(output: str, execution_only: bool) -> RunResults:
+    if execution_only:
+        tps = float(re.findall(r"Overall execution TPS: (\d+\.?\d*) txn/s", output)[-1])
+        gps = float(re.findall(r"Overall execution GPS: (\d+\.?\d*) gas/s", output)[-1])
```

Comment:
:trollface:  should really just print integers.

Reply:
we are printing integers in the aggregate table in the regression test. When you are doing single run, having more precision might be useful (or if you have workload that is in <100 tps.